libnmstate/error.py

```diff
@@ -101,3 +101,11 @@ class NmstateInternalError(NmstateError):
     """
 
     pass
+
+
+class NotSupportedByDevice(Exception):
```

Comment:
I am fine with adding the "Nmstate" prefix but I would keep the "ByDevice" suffix because Nmstate is supporting the features but the device not. What do you think?

Reply:
I meant NmstateNotSupportedByDeviceError

Reply:
To me, `NmstateNotSupportedError` does not imply to me  why it is not supported but leaves this open. When actually raising the exception, it would make sense to add the full reason as an argument, for example:

```
raise NmstateNotSupportedError(f"Interface '{interface}' does not support SR-IOV")
```

Also, if Nmstate would not be supporting a feature in general, it would probably be an `NmstateNotImplementedError`.

Reply:
That seems fine, thanks!